packages/zora/src/Zora.ts

```diff
@@ -363,6 +348,67 @@ export const simulateMint = async (
   }
 }
 
+export const simulateV2Mint = async (
+  mint: MintIntentParams,
+  value: bigint,
+  client: PublicClient,
+  account?: Address,
+): Promise<SimulateContractReturnType> => {
+  const { contractAddress, tokenId, amount, recipient, referral } = mint
+
+  const from = account ?? DEFAULT_ACCOUNT
+
+  const _tokenId = await getNextTokenId(client, contractAddress, tokenId)
+
+  await checkBytecode(client, contractAddress)
```

Comment:
No need to move this now but I could see this being generally useful - nice little portable helper.